### ISSUE TITLE
AbstractBigtableAdmin doesn't throw expected exceptions

### ISSUE THREAD

**Issue by heldtogether** [GoogleCloudPlatform/cloud-bigtable-client#1019]
Calling `createTable()` on any instance of `org.apache.hadoop.hbase.client.Admin` should throw various exceptions depending on why the table can't be created.

According to the docs in the interface:

```
@throws IllegalArgumentException if the table name is reserved
@throws org.apache.hadoop.hbase.MasterNotRunningException if master is not running
@throws org.apache.hadoop.hbase.TableExistsException if table already exists (If concurrent
    threads, the table may have been created between test-for-existence and attempt-at-creation).
@throws IOException
```

**Expected behavior:**
If table exists, throw `TableExistsException`.

**Observed behavior:**
If table exists, `IOException` is thrown.

Here is an example of the issue:

```java
    try {
        connection = BigtableConfiguration.connect("projectId", "instanceId");
        Admin admin = connection.getAdmin();

        HTableDescriptor descriptor = new HTableDescriptor(TableName.valueOf("tableName"));
        descriptor.addFamily(new HColumnDescriptor("cf"));
        
        admin.createTable(descriptor);
    } catch (TableExistsException e) {
        // This Exception is never thrown!
    } catch (IOException e) {
        // This Exception is always thrown!
    }
```

I can see the `IOException` is being thrown at https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java#L372 but can't trace it back to the original exception so can't submit a pull request.

**Comment by sduskis** [comment]
The Cloud Bigtable API uses gRPC infrastructure to send errors.  We currently don't have an easy way to send information about the specifics of the issue to ensure the expected behavior.

In this case, we'd need to send back a well defined HTTP error code, as well as protobuf describing the specifics of the problem (error_code = "FOUND", object_type="table", name = "table_name") or (error_code = "NOT_FOUND", object_type="instanceId", name = "instanceId") and etc.

We do not have a mechanism to send that type of information through our infrastructure at the beginning of the Cloud Bigtable project, so we decided to hold off from doing anything fancy and provide an error reason String.  The String can change over time.

In the long term, we need to fix this, but we don't have a timeline on this.

**Comment by heldtogether** [comment]
Can you suggest the best way of handling the exceptions, given the error reason string can change over time?

There's quite a big difference between what can be done with a `TableExistsException` (i.e. get the table instead and then further operations will work as expected) and any other `IOException` (i.e. there's a big problem contacting Bigtable, further operations won't work at all).

I'm cautious about calling `tableExists()` before `createTable()` as I might have multiple threads running this code and I think that introduces a race.

It would almost be better for `createTable()` to fail silently than throw any exception!

**Comment by heldtogether** [comment]
Fixed by #1025 and #1026 👍
